index.js

```diff
@@ -286,8 +295,14 @@ module.exports = function (filename, opts) {
     onReady,
 
     onDrain: onLoad(function (fn) {
-      if (Object.keys(blocksToBeWritten).length == 0) fn()
-      else waitingDrain.push(fn)
+      let blockIndexes = Object.keys(blocksToBeWritten)
+      if (blockIndexes.length == 0) fn()
+      else {
+        const latestBlockIndex = blockIndexes[blockIndexes.length-1]
+        let drains = waitingDrain[latestBlockIndex] || []
```

Comment:
Can merge without doing this, this is just a comment: `let` is for variables that are expected to be re-assigned. Objects and arrays that are mutated are not reassigned (we don't use `=` elsewhere), so in this case we could use `const drains`.

Reply:
Right, I still get confused by that. The whole stack vs reference thing from C++ is deeply ingrained in me ;-)